xocto/storage/storage.py

```diff
@@ -1107,6 +1126,158 @@ def fetch_file(self, key_path: str, version_id: str | None = None) -> StreamingB
     def fetch_file_contents(self, key_path: str, version_id: str | None = None) -> bytes:
         return self.fetch_file(key_path, version_id).read()
 
+    def fetch_file_contents_using_s3_select(
+        self,
+        key_path: str,
+        raw_sql: str,
+        input_serializer: s3_select.CSVInputSerializer | s3_select.ParquetInputSerializer,
+        output_serializer: s3_select.CSVOutputSerializer | s3_select.JSONOutputSerializer,
+        compression_type: s3_select.CompressionType | None = None,
+        scan_range: s3_select.ScanRange | None = None,
+        chunk_size: int | None = None,
+    ) -> Iterator[str]:
+        """
+        Localdev version of S3FileStore version. Does not support scan ranges.
+        """
+
+        if scan_range is not None:
+            raise NotImplementedError("Scan ranges are not supported for localdev")
+
+        input_file_path = self._filepath_for_key_path(key_path=key_path)
+
+        if isinstance(input_serializer, s3_select.CSVInputSerializer):
+            # https://docs.aws.amazon.com/AmazonS3/latest/API/API_CSVInput.html#AmazonS3-Type-CSVInput-QuoteCharacter
+            df = self.read_csv_with_serializer(
+                csv_file_path=input_file_path,
+                csv_input_serializer=input_serializer,
+                compression_type=compression_type,
+            )
+        elif isinstance(input_serializer, s3_select.ParquetInputSerializer):
+            parquet_file = parquet.ParquetFile(input_file_path)
+            df = parquet_file.read().to_pandas()
+
+        filtered_df = self.query_dataframe_with_sql(
+            raw_sql=raw_sql,
+            df=df,
+        )
+
+        if isinstance(output_serializer, s3_select.JSONOutputSerializer):
+            if output_serializer.RecordDelimiter != "\n":
+                raise NotImplementedError(
+                    "Record delimiter other than newline not implemented for JSON output in localdev"
+                )
+            result = filtered_df.to_json(orient="records", lines=True, date_format="iso")
+        elif isinstance(output_serializer, s3_select.CSVOutputSerializer):
+            result = self.output_csv_with_serializer(
+                df=filtered_df,
+                output_serializer=output_serializer,
+            )
+
+        yield result
+
+    def query_dataframe_with_sql(
+        self,
+        raw_sql: str,
+        df: pd.DataFrame,
+    ) -> pd.DataFrame:
+        con = duckdb.connect(database=":memory:")
+        con.register("s3_object", df)
```

Comment:
need to change this to s3object

Reply:
Would be better as a comment above this line, with a bit more detail on why you do this

Reply:
sorry this was just a reminder to me to change this.